lib/src/main/java/com/otaliastudios/transcoder/source/DefaultDataSource.java

```diff
@@ -88,6 +91,49 @@ public void initialize() {
         } */
     }
 
+    @Override
+    public ArrayList<Long> getKeyFrameTimestampsUs() {
+        return keyFrameTsUs;
+    }
+
+    private Boolean lastKeyFrame = false;
+    @Override
+    public long requestKeyFrameTimestamps() {
+
+        if(lastKeyFrame) return  -1L;
+        if(keyFrameTsUs.size() > 0) {
+            mExtractor.seekTo(keyFrameTsUs.get(keyFrameTsUs.size() - 1) + 1, MediaExtractor.SEEK_TO_NEXT_SYNC);
```

Comment:
Just realised ArrayList has a last() method.
Will make it much readable

Reply:
it's a kotlin method. not available in java @sahilbajaj